emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -0,0 +1,277 @@
+package edu.cornell;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import edu.cornell.emop.util.Util;
+import edu.illinois.starts.jdeps.DiffMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.diff.DiffFormatter;
+import org.eclipse.jgit.diff.Edit;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.treewalk.AbstractTreeIterator;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+import org.eclipse.jgit.treewalk.FileTreeIterator;
+import org.eclipse.jgit.util.io.DisabledOutputStream;
+
+@Mojo(name = "vms", requiresDirectInvocation = true, requiresDependencyResolution = ResolutionScope.TEST)
+@Execute(phase = LifecyclePhase.TEST, lifecycle = "vms")
+public class VmsMojo extends DiffMojo {
+
+    private final DiffFormatter diffFormatter = new DiffFormatter(DisabledOutputStream.INSTANCE);
+
+    // Represents the map between a file and a map representing the number of lines added or deleted at each line
+    // of the file. If the value is 0, the line has been modified in place.
+    // Note: If renames are involved, the old name of the file is used.
+    //          file        line     lines modified
+    private Map<String, Map<Integer, Integer>> lineChanges = new HashMap<>();
+
+    // Represents the renamed file's original name
+    private Map<String, String> renames = new HashMap<>();
+
+    // Describes new classes which have been made between commits
+    private Set<String> newClasses = new HashSet<>();
+
+    private Set<List<String>> oldViolations;
```

Comment:
Not super necessary but a good to have would be to briefly comment on the indices of each List (it seems like a triple between spec, class, line number?)

Reply:
or you could even refactor to create a new class `Violation`

Reply:
Made a new `Violation` class :)